### ISSUE TITLE
CVEs in optional components (kiali, grafana, prometheus, jaeger)

### ISSUE THREAD

**Issue by kramvan1** [istio/istio#20659]
**Bug description**

CVEs in optional components:
```
kiali/kiali:v1.4 CVE-2016-10739,CVE-2018-0495,CVE-2018-0734,CVE-2018-12404,CVE-2018-14618,CVE-2018-16062,CVE-2018-16402,CVE-2018-16403,CVE-2018-16842,CVE-2018-18310,CVE-2018-18520,CVE-2018-18521,CVE-2018-20532,CVE-2018-20533,CVE-2018-20534,CVE-2019-11729,CVE-2019-11745,CVE-2019-1559,CVE-2019-3858,CVE-2019-3861,CVE-2019-3862,CVE-2019-7149,CVE-2019-7150,CVE-2019-7664,CVE-2019-7665
grafana/grafana:6.1.6 CVE-2019-1543,CVE-2019-1547,CVE-2019-1549,CVE-2019-1563,CVE-2019-5094
```

**Expected behavior**

Need to have optional components updated for CVEs with the regular patch updates so they don't become security exposures.

**Steps to reproduce the bug**

Run CVE scan against the images above (should already be part of build process)

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

1.3 and 1.4, ANY release need to keep up with security patches

**How was Istio installed?**

N/A

**Environment where bug was observed (cloud vendor, OS, etc)**

N/A

N/A

**Comment by kramvan1** [comment]
cc @linsun @rvennam @ericvn

**Comment by howardjohn** [comment]
These components are intended for demo only and should not be considered production ready

**Comment by howardjohn** [comment]
cc @dgn @fpesce

**Comment by howardjohn** [comment]
Also note that there is no 6.1.7 build or anything of grafana. I don't think we should be updating minor releases of dependencies during unless they directly impact Istio's security. this falls under the same category of issues we ran into with Envoy where our support windows our longer than our underlying dependencies

**Comment by kramvan1** [comment]
@howardjohn Thx for the feedback here. We will move forward with a new plan for using the extras and prometheus.

**Comment by sdake** [comment]
I don't agree with you @howardjohn. While I recognize its painful to release these dependencies, our userbase expects a CVE-free product - all the time. In some cases, this just isn't viable, but I really think we can focus on the area of addons and come to a consensus conclusion.

**Comment by sdake** [comment]
@kramvan1 while I don't agree with @howardjohn , I do agree with some aspects of his position, especially around supportability of the addons in production environments. We should be exceedingly clear that the addons are for evaluation only - once we have a good set of documentation that describes how to integrate your own addons with Istio. But we need that before we can take any further steps.

I would encourage you to investigate your own production deployments of the addons - as what we have today is not production-ready.

**Comment by linsun** [comment]
Note: we are using a much newer version of Kiali and grafana in master

**Comment by JimmyCYJ** [comment]
Assign to Limin for triage. Perhaps we need a plan to support those addons

**Comment by jshaughn** [comment]
We try to update Kiali for each minor release of Istio. Kiali is slated to be updated for Istio 1.5 https://github.com/istio/istio/pull/20840.   If it is decided that you need Kiali updated for prior release streams we can discuss how to best handle that.  Perhaps open a Kiali issue?

**Comment by incfly** [comment]
need some investigation on the status quo, might be a good potential WG meeting topic.

**Comment by howardjohn** [comment]
related:
https://docs.google.com/document/d/1jJ4kkALCUPWIlz_Ldu6jpeelMNxyY_SZDMaZmeSpfkM

**Comment by myidpt** [comment]
I think this can be resolved by the production security WG.
@fpesce

**Comment by fpesce** [comment]
@myidpt exact, we probably need to setup a similar approach than Envoy's security posture of their extensions. (c.f. https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/threat_model#core-and-extensions )

**Comment by JimmyCYJ** [comment]
@fpesce @myidpt Do we need a separate label for these issues?

**Comment by fpesce** [comment]
@JimmyCYJ yes. Also, feel free to add this point to the next PS WG meeting notes agenda.

**Comment by kramvan1** [comment]
Will close this as monitoring addons are going away.
